extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java

```diff
@@ -143,6 +149,40 @@ public void updateClasspath(IProject project, IProgressMonitor monitor) throws C
         }
     }
 
+    /**
+     * Update the project dependencies of the project.
+     * @throws CoreException
+     */
+    public void addProjectDependencies(IProject project, IProgressMonitor monitor) throws CoreException {
+        IPath rootPath = ProjectUtils.findBelongedWorkspaceRoot(project.getLocation());
+        if (rootPath == null) {
+            JavaLanguageServerPlugin.logError("Cannot find workspace root for project: " + project.getName());
+            return;
+        }
+        BuildServer buildServer = ImporterPlugin.getBuildServerConnection(rootPath);
+        List<BuildTarget> buildTargets = Utils.getBuildTargetsByProjectUri(buildServer, project.getLocationURI());
+        Set<BuildTargetIdentifier> projectDependencies = new HashSet<>();
+        for (BuildTarget buildTarget : buildTargets) {
+            projectDependencies.addAll(buildTarget.getDependencies());
```

Comment:
why does `buildTarget.getDependencies()` represent the project dependencies? Is it possible that the dependency is also a jar dependency?

Reply:
It won't. According to the spec, the dependencies of a build target are `The direct upstream build target dependencies of this build target`.

For those jar dependencies, they are calculated in another request [buildTarget/dependencyModules](https://build-server-protocol.github.io/docs/specification#buildtargetdependencymodules-request)